src/Service/Parsers/ODataASTVisitor.cs

```diff
@@ -37,6 +37,17 @@ public override string Visit(BinaryOperatorNode nodeIn)
             // In order traversal but add parens to maintain order of logical operations
             string left = nodeIn.Left.Accept(this);
             string right = nodeIn.Right.Accept(this);
+
+            if (nodeIn.Left.GetType() == typeof(SingleValuePropertyAccessNode) && nodeIn.Right.GetType() == typeof(ConstantNode) ||
+                nodeIn.Left.GetType() == typeof(ConstantNode) && nodeIn.Right.GetType() == typeof(SingleValuePropertyAccessNode))
+            {
+                SingleValuePropertyAccessNode propertyNode = nodeIn.Left.GetType() == typeof(SingleValuePropertyAccessNode) ?
+                    (SingleValuePropertyAccessNode)nodeIn.Left : (SingleValuePropertyAccessNode)nodeIn.Right;
+                string? paramName = $"{BaseQueryStructure.PARAM_NAME_PREFIX}param{_struct.Counter.Current() - 1}";
+                _metadataProvider.TryGetBackingColumn(_struct.EntityName, propertyNode.Property.Name, out string? backingColumnName);
+                _struct.Parameters[paramName] = new(_struct.Parameters[paramName].Item1, _struct.GetUnderlyingSourceDefinition().Columns[backingColumnName!].DbType);
```

Comment:
how can we guarantee _struct.Parameters[paramName] is already present? Are we for some reason overwriting the old object with the new one? reusing just the first Item?